packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Snippets/BinaryContentSnippets.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.Generator.CSharp.ClientModel.Snippets
     internal static class BinaryContentSnippets
     {
         public static ScopedApi<BinaryContent> Create(ValueExpression serializable)
-            => Static<BinaryContent>().Invoke(nameof(BinaryContent.Create), serializable).As<BinaryContent>();
+            => Static(ClientModelPlugin.Instance.TypeFactory.RequestContentApi.RequestContentType).Invoke(nameof(BinaryContent.Create), serializable).As<BinaryContent>();
```

Comment:
I think we should not change this.
These snippets for the type `BinaryContent` therefore everything should be only for this type.
If in a plugin we replaced this type with `RequestContent`, this would produce wrong code.
Potentially maybe we no longer need this file, because anywhere we are using BinaryContent should be extensible (for example changed to RequestContent in a plugin), therefore those APIs should come from a `RequestContentApi`

Reply:
You are right!
Renamed this to RequestContentApiSnippets and update the return type accordingly.